test_plugin/tests/test.js

```diff
@@ -39,20 +39,23 @@ function runTestSync() {
   console.log(`Plugin Sync Response: ${textDecoder.decode(response)}`);
 }
 
-Deno.core.setAsyncHandler(testAsync, (response) => {
-  console.log(`Plugin Async Response: ${textDecoder.decode(response)}`);
-});
-
 function runTestAsync() {
-  const response = Deno.core.dispatch(
-    testAsync,
-    new Uint8Array([116, 101, 115, 116]),
-    new Uint8Array([116, 101, 115, 116])
-  );
-
-  if (response != null || response != undefined) {
-    throw new Error("Expected null response!");
-  }
+  return new Promise((res, rej) => {
+    Deno.core.setAsyncHandler(testAsync, (response) => {
```

Comment:
Don't know if this is the kind of thing that should be set only once, but this enables that the async test can be awaited.